src/IonBinaryWriter.ts

```diff
@@ -194,13 +197,18 @@ export class BinaryWriter extends AbstractWriter {
     this.addNode(new BytesNode(this.writer, this.getCurrentContainer(), IonTypes.FLOAT, this.encodeAnnotations(this._annotations), bytes));
   }
 
-  writeInt(value: number) : void {
+  writeInt(value: number | JSBI | null) : void {
```

Comment:
perhaps a typealias for `value`?

Reply:
What alias would you suggest? I'm a little afraid to introduce an important-sounding type named `Integer` or `IonInteger`.

Reply:
By type alias I mean http://www.typescriptlang.org/docs/handbook/advanced-types.html#type-aliases

Mostly a rename to aid readability, `MaybeInt` -- not sure if there is a `Maybe<X>` or `Option<X>` and how it is implemented but generics might make things more complex